src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -0,0 +1,1126 @@
+  {
+    FaceInfo f = allFaces[i];
+    FaceInfo f1 = allFaces[i+1];
+
+    // If two successive faces are the same - this is an interior facet
+    if( f.v[0]==f1.v[0] && f.v[1]==f1.v[1] && f.v[2]==f1.v[2] )
+    {
+      m_allFacesToNeighbors[f.v[3]] = f1.v[3];
+      m_allFacesToNeighbors[f1.v[3]] = f.v[3];
+      m_uniqueFaces[curFace] = f.v[3];
+      curFace++;
+      ++i;
+    }
+    // If not this is a boundary face
+    else
+    {
+      m_allFacesToNeighbors[f.v[3]] = NO_ID;
+      m_uniqueFaces[curFace] = f.v[3];
+      m_isBoundaryFace[curFace] = true;
+      curFace++;
+    }
+  }
+  // Last facet is a boundary facet 
+  if (i < nbTotalFaces) 
+  {
+    FaceInfo f = allFaces[i];
+    m_allFacesToNeighbors[f.v[3]] = NO_ID;
+    m_uniqueFaces[curFace] = allFaces[i].v[3];
+    m_isBoundaryFace[curFace] = true;
+    curFace++;
+  }
+  // Set the number of faces 
+  nbFaces = curFace; 
+  // Remove the non-filled values 
+  m_uniqueFaces.resize( nbFaces );
+  m_isBoundaryFace.resize( nbFaces );
+
+  return true;
+}
+
+/**
+ * @brief Compute all the edges of the cells of the cell blocks
+ * Fills the edge information 
+ * 
+ * TODO CellType dependant 
+ */
+bool HexMeshConnectivityBuilder::computeEdges()
+{
+  m_allEdges.resize( 0 );
+  m_uniqueEdges.resize( 0 );
+
+  // 1 - Allocate 
+  SizeOfStuff nbAllEdges = nbElements * Hex::nbEdges;
+  m_allEdges.resize( nbAllEdges );
+  
+  // 2 - Get all edges
+  localIndex cur = 0 ;
+  for (unsigned int i = 0; i < nbCellBlocks(); ++i)
+  {
+    CellBlock const & block = getCellBlock(i);
+    CellVertexIndices const & cells = block.getElemToNodes();
+
+    for (int j = 0; j < cells.size(0); ++j)
+    {
+      for(unsigned int e = 0; e < Hex::nbEdges; ++e)
+      {
+        assert (cur < nbAllEdges );
+
+        LocalVertexIndex v0 = cells( j, Hex::edgeVertex[e][0]);
+        LocalVertexIndex v1 = cells( j, Hex::edgeVertex[e][1]);
+
+        if (v1 < v0 ) { std::swap(v0, v1); }
+        LocalVertexIndex v = v0 * nbNodes + v1;
+        LocalEdgeIndex id = j * Hex::nbEdges + e;
+
+        m_allEdges[cur] = EdgeInfo(v, id);
+        ++cur;
+      }
+    }
+  }
+  // 3 - Sort according to vertices 
+  std::sort( m_allEdges.begin(), m_allEdges.end(), compareEdgeInfo );
+  
+  // 4 - Reserve space unique edges 
+  // If a CellBlockManager manages a connected set of cells (any type of cell)
+  // bounded by a sphere nbNodes - nbEdges + nbFaces - nbElements = 1 
+  // For Hexes: 12 * nbCells = 4 * nbEdges + BoundaryEdges + Singularities - speculation 3.5 factor
+  SizeOfStuff guessNbEdges = nbFaces != 0 ? nbNodes + nbFaces - nbElements : 3.5 * nbElements;
+  m_uniqueEdges.reserve( guessNbEdges );
+
+  // 4 - Get unique edges
+  unsigned int i = 0;
+  while( i < m_allEdges.size() )
+  {
+    m_uniqueEdges.push_back(i);
+    unsigned int j = i+1;
+    while( j < m_allEdges.size() && equalEdgeInfo(m_allEdges[i], m_allEdges[j]) )
+    { 
+      ++j; 
+    }
+    i = j;
+  }
+  nbEdges = m_uniqueEdges.size();
```

Comment:
Is there a reason to store `nbEdges` instead of returning `m_uniqueEdges.size()` since we also store `m_uniqueEdges`?

Reply:
A history of copy-paste from pre-existing code and consistency sith the storage of numNodes. 
Removed.